microprofile/grpc/metrics/src/main/java/io/helidon/microprofile/grpc/metrics/GrpcMetricAnnotationDiscoveryObserverImpl.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.microprofile.grpc.metrics;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+
+import io.helidon.microprofile.grpc.core.AnnotatedMethod;
+import io.helidon.microprofile.grpc.core.GrpcMethod;
+import io.helidon.microprofile.metrics.MetricAnnotationDiscovery;
+import io.helidon.microprofile.metrics.spi.MetricAnnotationDiscoveryObserver;
+
+import jakarta.enterprise.inject.spi.configurator.AnnotatedMethodConfigurator;
+
+/**
+ * The gRPC implementation of {@link io.helidon.microprofile.metrics.spi.MetricAnnotationDiscoveryObserver}.
+ *
+ * <p>
+ *     This implementation
+ * </p>
+ */
+public class GrpcMetricAnnotationDiscoveryObserverImpl implements MetricAnnotationDiscoveryObserver {
+
+    private static GrpcMetricAnnotationDiscoveryObserverImpl instance;
+
+    static GrpcMetricAnnotationDiscoveryObserverImpl instance() {
+        return instance;
+    }
+
+    private static void instance(GrpcMetricAnnotationDiscoveryObserverImpl value) {
+        instance = value;
+    }
+
+    private final Map<Method,
+            Map<Class<? extends Annotation>, MetricAnnotationDiscovery.OfMethod>> discoveriesByMethod =
+            new HashMap<>();
+
+    /**
+     * Creates a new instance.
+     */
+    public GrpcMetricAnnotationDiscoveryObserverImpl() {
+        instance(this);
```

Comment:
This escape of instance to a static field is dangerous. Couldn't this be solved by using static fields to store the information, rather then storing the whole instance to a static field?

Reply:
Done